staging/src/k8s.io/legacy-cloud-providers/azure/azure.go

```diff
@@ -138,16 +138,8 @@ type Config struct {
 	//   CloudProviderBackoffJitter are omitted.
 	// "default" will be used if not specified.
 	CloudProviderBackoffMode string `json:"cloudProviderBackoffMode,omitempty" yaml:"cloudProviderBackoffMode,omitempty"`
-	// Enable rate limiting
-	CloudProviderRateLimit bool `json:"cloudProviderRateLimit,omitempty" yaml:"cloudProviderRateLimit,omitempty"`
-	// Rate limit QPS (Read)
-	CloudProviderRateLimitQPS float32 `json:"cloudProviderRateLimitQPS,omitempty" yaml:"cloudProviderRateLimitQPS,omitempty"`
-	// Rate limit Bucket Size
-	CloudProviderRateLimitBucket int `json:"cloudProviderRateLimitBucket,omitempty" yaml:"cloudProviderRateLimitBucket,omitempty"`
-	// Rate limit QPS (Write)
-	CloudProviderRateLimitQPSWrite float32 `json:"cloudProviderRateLimitQPSWrite,omitempty" yaml:"cloudProviderRateLimitQPSWrite,omitempty"`
-	// Rate limit Bucket Size
-	CloudProviderRateLimitBucketWrite int `json:"cloudProviderRateLimitBucketWrite,omitempty" yaml:"cloudProviderRateLimitBucketWrite,omitempty"`
```

Comment:
Removing those options would break existing clusters, e.g. upgrading from old releases. Could we keep them as the default values if azure client's rate limit is not configured?

Reply:
I don't think it would, they just would be ignored, wouldn't they ?

Reply:
"ignored" is the issue.  We shouldn't ignore the configured options silently when customer is doing cluster upgrade. Could you add them back and set them as the default values for cloudProviderRateLimiting.

Reply:
I don't like this, we shouldn't keep history if not relevant anymore just for convenience because if we were to change it again and again it would become unbearable.

Reply:
but this is not history, it is configure option. And it also provides an easy way to setup default values. 

Since it is configure options, we shouldn't change it again and again. Why do you think it need be changed later?

Reply:
Well we are changing them now. I don't see why there would be no reasons to change them again in the future.

We are already enforcing default values in the code, I don't think we should allow overriding default values.

If users want to change values they should use the new ones we are defining right now.

Reply:
we does allow customers to override the default values